app/actions/remote/custom_profile.ts

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {forceLogoutIfNecessary} from '@actions/remote/session';
+import DatabaseManager from '@database/manager';
+import NetworkManager from '@managers/network_manager';
+import {getFullErrorMessage} from '@utils/errors';
+import {logDebug} from '@utils/log';
+
+import type {CustomProfileField, CustomAttribute, CustomAttributeSet, CustomProfileAttribute, UserCustomProfileAttributeSimple} from '@typings/api/custom_profile_attributes';
+
+/**
+ * Fetches custom profile fields from the server
+ * @param serverUrl - The server URL
+ * @returns Promise with the custom profile fields or error
+ */
+export const fetchCustomProfileFields = async (serverUrl: string): Promise<{fields?: CustomProfileField[]; error?: unknown}> => {
+    try {
+        const client = NetworkManager.getClient(serverUrl);
+        const {operator} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+        const fields = await client.getCustomProfileAttributeFields();
```

Comment:
so, we'll be fetching all fields every time? I think this can be improved to only fetched those fields that have been modified/deleted since a moment in time.

If that is not possible today, it should be included in the API. 5/5

Reply:
it is not possible today, but it is something the team is aware.